emcc.py

```diff
@@ -516,6 +515,44 @@ def get_all_js_syms():
     if shared.is_c_symbol(name):
       name = shared.demangle_c_symbol_name(name)
       library_syms.add(name)
+  return library_syms
+
+
+@ToolchainProfiler.profile_block('JS symbol generation')
+def get_all_js_syms():
+  # To avoid the cost of calling generate_js_symbols each time an executable is
+  # linked we cache symbol lists for the N most recently used configs.
+  # We define a cache hit as when the settings and `--js-library` contents are
+  # identical.
+  input_data = json.dumps(settings.dict(), sort_keys=True, indent=2) + '\n'
+  for jslib in settings.JS_LIBRARIES:
+    if os.path.abspath(jslib):
+      jslib = utils.path_from_root('src', jslib)
+    input_data += read_file(jslib)
```

Comment:
Given the speed you measured this is probably fast enough, but this concatenates potentially large strings here. It could be more efficient to accumulate an array of strings.

Reply:
Done